src/components/FileList/treeTable.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react'
+import ThCustomComponent from './ThCustomComponent'
+import TrCustomComponent from './TrCustomComponent'
+import TdCustomComponent from './TdCustomComponent'
+import PivotCustomComponent from './PivotCustomComponent'
+
+/**
+ * TreeTable HOC Component
+ * This is a custom implementation based on react-table TreeTable HOC
+ * The purpose is to provide the data to the renderer, excluding any "undefined" 
+ * row generated by the Pivot grouping function
+ * 
+ * It renders custom components instead the default of ReactTable
+ * 
+ */
+export default Component => {
+  const wrapper = class RTTreeTable extends React.Component {
+    constructor(props) {
+      super(props)
+      this.getWrappedInstance.bind(this)
+      this.getTrProps.bind(this)
+      this.getSubrows.bind(this)
+    }
+
+    // this is so we can expose the underlying ReactTable to get at the sortedData for selectAll
+    getWrappedInstance = () => {
+      if (!this.wrappedInstance) console.warn('RTTreeTable - No wrapped instance')
+      if (this.wrappedInstance.getWrappedInstance) return this.wrappedInstance.getWrappedInstance()
+      else return this.wrappedInstance
+    }
+
+    /**
+     * This is the custom implementation where each subRow with _pivotVal "undefined" is parsed
+     * in any of his _subRows childs, until find the first _pivotVal !== "undefined"
+     * 
+     * Once the object has been found, than it will returned to the first parent, 
+     * updating the value of _nestingLevel as well
+     * 
+     */
+    getSubrows = subRows => {
+      subRows && subRows.forEach((subRow, index) => {
+        if (subRow._pivotVal === "undefined" && subRow._subRows) {
```

Comment:
So the value of the string will be "undefined"? or might the value be `undefined`? Just making sure! 😜

Reply:
yes, "undefined" is the value of the string. This is a fix i implemented to solve a wrong behaviour of react-table that generates some "undefined" folders